### PR TITLE
[SYCL][UR][L0] Fetch Level Zero adapter source from the Unified Runtime repo

### PR DESCRIPTION
**Pull request by callumfare**
This is the second part of moving the Level Zero adapter to the Unified Runtime repo, after https://github.com/oneapi-src/unified-runtime/pull/879

### CODE REVIEW

File: sycl/plugins/level_zero/CMakeLists.txt [hunk 1]
```diff
@@ -93,39 +75,39 @@ add_sycl_plugin(level_zero
...
+    "${UNIFIED_RUNTIME_SOURCE_DIR}/source/adapters/level_zero/event.cpp"
+    "${UNIFIED_RUNTIME_SOURCE_DIR}/source/adapters/level_zero/image.cpp"
+    "${UNIFIED_RUNTIME_SOURCE_DIR}/source/adapters/level_zero/kernel.cpp"
+    "${UNIFIED_RUNTIME_SOURCE_DIR}/source/adapters/level_zero/memory.cpp"
+    "${UNIFIED_RUNTIME_SOURCE_DIR}/source/adapters/level_zero/platform.cpp"
+    "${UNIFIED_RUNTIME_SOURCE_DIR}/source/adapters/level_zero/program.cpp"
+    "${UNIFIED_RUNTIME_SOURCE_DIR}/source/adapters/level_zero/queue.cpp"
+    "${UNIFIED_RUNTIME_SOURCE_DIR}/source/adapters/level_zero/sampler.cpp"
+    "${UNIFIED_RUNTIME_SOURCE_DIR}/source/adapters/level_zero/usm.cpp"
+    "${UNIFIED_RUNTIME_SOURCE_DIR}/source/adapters/level_zero/usm_p2p.cpp"
```
Comment (kbenzie) [inline sycl/plugins/level_zero/CMakeLists.txt:110]: Ideally I think we should get this list of source files from UR, either in a variable we define or by querying the list of source files from the L0 adapter target. That way we won't need to update this list if/when a new file is added to the L0 target in UR.

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 2]
```diff
@@ -3,9 +3,11 @@
 if (NOT DEFINED UNIFIED_RUNTIME_LIBRARY OR NOT DEFINED UNIFIED_RUNTIME_INCLUDE_DIR)
   include(FetchContent)
 
-  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
-  set(UNIFIED_RUNTIME_TAG b3cc9ae3f99ca7faff1ba765dd36652fef2cfddd)
+  # TODO: Update when L0 adapter source is merged
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:6]: https://github.com/oneapi-src/unified-runtime/pull/879

**Comment by callumfare** [comment]
The CI failure is because the other adapters are out-of-sync with the UR commit now. This will be fixed by rebasing my test UR branch after https://github.com/oneapi-src/unified-runtime/pull/881 goes in

File: sycl/plugins/level_zero/CMakeLists.txt [hunk 3]
```diff
@@ -44,24 +44,6 @@ if (NOT DEFINED LEVEL_ZERO_LIBRARY OR NOT DEFINED LEVEL_ZERO_INCLUDE_DIR)
     ${level-zero-loader_SOURCE_DIR}/include CACHE PATH "Path to Level Zero Headers")
 endif()
 
-add_library (LevelZeroLoader INTERFACE)
```
Comment (jandres742) [inline sycl/plugins/level_zero/CMakeLists.txt:47]: comment here https://github.com/oneapi-src/unified-runtime/pull/879/files#r1334743502 about the L0 loader tag.

File: sycl/plugins/unified_runtime/pi2ur.hpp [hunk 4]
```diff
@@ -1254,8 +1254,9 @@ inline pi_result piDeviceGetInfo(pi_device Device, pi_device_info ParamName,
     PI_TO_UR_MAP_DEVICE_INFO(
         PI_EXT_ONEAPI_DEVICE_INFO_INTEROP_SEMAPHORE_EXPORT_SUPPORT,
         UR_DEVICE_INFO_INTEROP_SEMAPHORE_EXPORT_SUPPORT_EXP)
-    PI_TO_UR_MAP_DEVICE_INFO(PI_EXT_INTEL_DEVICE_INFO_ESIMD_SUPPORT,
-                             UR_DEVICE_INFO_ESIMD_SUPPORT)
+    // TODO: Re-enable when UR 'adapters' branch is updated
```
Comment (jandres742) [inline sycl/plugins/unified_runtime/pi2ur.hpp:1257]: why do we need this change?

Comment (callumfare) [reply sycl/plugins/unified_runtime/pi2ur.hpp:1257]: This is just a temporary fix until the `adapters` branch in the UR repo is up to date, currently it's quite a few commits behind `main` so is missing this enum value (see https://github.com/oneapi-src/unified-runtime/pull/881). I'm planning on keeping this in draft until that is resolved and the L0 adapter source is actually merged.

Comment (callumfare) [reply sycl/plugins/unified_runtime/pi2ur.hpp]: This has been reverted now

**Review by Seanst98 — APPROVED** [review]

**Comment by kbenzie** [comment]
@intel/llvm-reviewers-runtime @intel/dpcpp-l0-pi-reviewers please review ASAP.

**Review by smaslov-intel — APPROVED** [review]
LGTM

**Review by jandres742 — APPROVED** [review]

**Review by bso-intel — APPROVED** [review]

**Comment by kbenzie** [comment]
@intel/llvm-gatekeepers this is ready to merge.

**Pull request merged**